### DISCUSSION TITLE
CSS Compiler Issue

### DISCUSSION THREAD

**Discussion by shaq-tff** [teambit/bit#6727] [Help] [answered]
Hello everyone, I just started using Bit and been struggling with the issue I'm facing with compiler. I've published a Bit component and tried to use it on a NextJS App , but I'm getting error on CSS. I suspect that it is because the built in compiler of Bit in react environment.

Here's the error when I import a SCSS Modules:
<img width="1019" alt="Screen Shot 2022-12-01 at 7 25 09 PM" src="https://user-images.githubusercontent.com/87931842/205041093-c13c9001-ad36-478a-9660-bd6871fb4343.png">

Here are the compiled files:
<img width="470" alt="Screen Shot 2022-12-01 at 6 46 25 PM" src="https://user-images.githubusercontent.com/87931842/205041440-915843d3-69ea-402a-9b76-b894ce7e5201.png">

Here's how I imported the Bit component:
<img width="480" alt="Screen Shot 2022-12-01 at 7 35 18 PM" src="https://user-images.githubusercontent.com/87931842/205043069-bb4e7eaf-6588-4e4d-9597-bb725d22c715.png">

At first I thought it was CSS Modules that's producing the error, so I tried to make it just plain css. But it gives this new error:
<img width="987" alt="Screen Shot 2022-12-01 at 6 45 10 PM" src="https://user-images.githubusercontent.com/87931842/205051833-cfc87875-9f41-49ac-823b-b1adfdeb7f80.png">

I was just wondering if I did something wrong or it's just that compiling CSS is not a built-in functionality of Bit compiler?

By the way, the component that I'm testing is in React Environment:
<img width="525" alt="Screen Shot 2022-12-01 at 8 34 00 PM" src="https://user-images.githubusercontent.com/87931842/205054104-e95d4108-681f-414a-acc4-2000140e9050.png">

**Comment by benjgil** [comment, marked as answer]
hi @shaq-tff 

tl;dr - these are actually nextjs things - e.g. [here](https://github.com/vercel/next.js/issues/19936). Nextjs doesnt compile things from node_modules (and also wont accept global css from there if you're just flat importing a css file in a component). 
The best solution if you still want to use scss with your bit components is to get nextjs to compile them using the [next-transpile-modules](https://www.npmjs.com/package/next-transpile-modules) package.

More detail:
Bit doesnt compile your css, so that the consuming application will compile it using it's own minification etc strategy. So in this case if your app could use sass @use etc functions/mixins and only convert to css at the bundling stage.
I do however have a task open to create a css compilation task, for those who need it, so that might be available soon. But i cant say when i'll get to it, so it's best to go for next-tm for now as above

**Reply by shaq-tff** [reply]
Hello @benjgil, thank you so much for helping out! It now works on NextJS, however, I tried it on Shopify Hydrogen and it produces the same error. Hydrogen uses Vite for compiler, for that to fix, I needed to do something similar in Vite config, right?

By the way, thank you for the info about Bit not compiling scss. I agree with that approach, that's more powerful since we could consume some mixins / placeholders / variables, etc. via `@use`

**Reply by benjgil** [reply]
Yep, I would guess you'll need to add the same to Vite's config like you said.
